Lesson19/src/slideshow.js

```diff
@@ -0,0 +1,245 @@
+const slideList = document.getElementById("js-slide-list");
+const slideWrapper = document.getElementById("js-slide");
+
+const createElementWithClassName = (type, className) => {
+  const element = document.createElement(type);
+  element.className = className;
+  return element;
+};
+
+const createLoading = () => {
+  const loading = createElementWithClassName("img", "loading");
+  loading.id = "js-loading";
+  loading.src = "./img/loading-circle.gif";
+  return loading;
+};
+
+const removeLoading = () => document.getElementById("js-loading").remove();
+
+const createErrorMessage = (error) => {
+  const errorMessage = createElementWithClassName("p", "error-message");
+  errorMessage.textContent = error;
+  return errorMessage;
+};
+
+const fetchErrorHandling = async (response) => {
+  if (response.ok) {
+    return await response.json();
+  } else {
+    throw new Error("サーバーエラーが発生しました");
+  }
+};
+
+const getJsonOrError = async (url) => {
+  const response = await fetch(url);
+  const json = await fetchErrorHandling(response);
+  return json;
+};
+
+const getSlideImgData = async () => {
+  return new Promise((resolve) => {
+    setTimeout(
+      () => resolve(getJsonOrError("https://myjson.dit.upm.es/api/bins/ckn1")),
+      3000
+    );
+  });
+};
+
+const tryGetSlideImgData = async () => {
+  slideWrapper.appendChild(createLoading());
+  try {
+    const slideImgData = await getSlideImgData();
+    if (slideImgData.length === 0) {
+      throw new Error("No data...");
+    }
+    return slideImgData;
+  } catch (e) {
+    slideWrapper.appendChild(createErrorMessage(e));
+  } finally {
+    removeLoading();
+  }
+};
+
+const init = async () => {
+  const imgData = await tryGetSlideImgData();
+  if (imgData) {
+    initSlideItem(imgData);
+    initArrowButtons();
+    initIndicator(imgData.length);
+    slideWrapper.appendChild(createCounterOfSlide(imgData));
+    autoPlay.start();
+  }
+};
+
+init();
+
+/**
+ * index of the element being displayed.
+ * @type {Number}
+ */
+let currentIndex = 0;
+
+const playOfSlideshow = () => {
+  switchIndicator();
+  switchSlideImg();
+  updateOfCounter();
+  toggleTheDisabled();
+}
+
+const autoPlay = function () {
+  let autoPlayerId;
+  const start = () => {
+    autoPlayerId = setInterval(() => {
+      const slideItems = [...document.querySelectorAll(".slide__item")];
+      currentIndex < slideItems.length - 1 ? ++currentIndex : currentIndex = 0;
+      playOfSlideshow();
+    }, 3000)
+  }
+  const reset = () => {
+    clearInterval(autoPlayerId);
+    start();
+  }
+  return {
+    start: start,
+    reset: reset
+  }
```

Comment:
nit

```suggestion
  return {
    start,
    reset
  }
```

Reply:
Thanks for the review.

I thought it would be easier to understand if I simply returned what I stored in the variable.
Please check back as we have fixed this.
7ec6739